documentation/API-design-guidelines.md

```diff
@@ -1381,39 +1411,65 @@ The Following table provides `/subscriptions` attributes
+| types | string | Type of event subscribed. This attribute **must** be present in the `POST` request. It is required by API project to provide an enum for this attribute. `type` must follow the format: `org.camaraproject.<api-name>.<api-version>.<event-name>` with the `api-version` with letter `v` and the major version like  ``org.camaraproject.device-roaming-subscriptions.v1.roaming-status`` - Note: An array of types could be passed **but as of now only one value MUST passed**. Use of multiple value will be open later at API level.| mandatory  |
+| config | object | Implementation-specific configuration parameters needed by the subscription manager for acquiring events. In CAMARA we have predefined attributes like ``subscriptionExpireTime``, ``subscriptionMaxEvents`` or ``initialEvent``. see detail below.| mandatory |
+| id | string | Identifier of the event subscription - This attribute must not be present in the POST request as it is provided by API server | mandatory in server response |
 | startsAt | string - date-time| Date when the event subscription will begin/began. This attribute must not be present in the `POST` request as it is provided by API server. It must be present in `GET` endpoints | optional |
-| expiresAt | string - date-time| Date when the event subscription will expire. This attribute must not be present in the `POST` request as it is provided by API server.  | optional |
-| subscriptionMaxEvents | integer | Identifies the maximum number of event reports to be generated (>=1) - Once this number is reached, the subscription ends | optional |
-| subscriptionDetail | object | Object defined for each event subscription depending on the event - it could be for example the ueID targeted by the subscription | optional |
+| expiresAt | string - date-time| Date when the event subscription will expire. This attribute must not be present in the `POST` request as it is provided (optionally) by API server.  | optional |
+| status | string | Current status of the subscription - Management of Subscription state engine is not mandatory for now. Note: not all statuses may be considered to be implemented. See below status table. | optional |
 
 
-The `webhook` object definition: 
 
-| name | type | attribute description | cardinality |
-| ----- |	-----  |	 -----  |  -----  | 
-| notificationUrl | string | https callback address where the event notification must be POST-ed | mandatory |
-| notificationAuthToken | string | OAuth2 token to be used by the callback API endpoint. It MUST be indicated within HTTP Authorization header e.g. Authorization: Bearer $notificationAuthToken  | optional |
+``sinkCredential`` attributes table (must be only access token for now):
 
-The `subscriptionDetail` must have at least a type attribute:
+| attribute name | type | attribute description | cardinality |
+| ----- |	-----  |	 -----  | -----  | 
+| credentialtype | string | Type of the credential - MUST be set to `ACCESSTOKEN` for now | mandatory |
```

Comment:
It has been followed https://github.com/cloudevents/spec/blob/main/subscriptions/spec.md#323-sink-credentials to be compliant with CloudEvent Model. And agreed to use so far "ACCESSTOKEN" (FYI)

Reply:
Edited: replaced SHARED_SECRET by PLAIN
Please see other comment about Camara becoming a client of the client. Is that what was agreed upon in Commonalities? How is the sinkCredentials created and know to Camara?

Reply:
CloudEvents does not specify anythink about security
> The CloudEvents spec currently does not mandate or advocate any specific mechanism or principles in the matter of **Authorization**, Data Integrity and Confidentiality, as the current intention behind this spec is not to define Security principles with regards to CloudEvents. 

In light of that I see that ACCESS_TOKEN can be correct, because of Commonalities can define whatever we want in regard of **Authorization**.

What is the subscription section and the PR is then missing, is how the access token is created and what the sink MUST do to validate it.

Reply:
@PedroDiez I think ACCESS_TOKEN is only OK if the AccessToken is created by Camara and validated by sink. There is no documentation about token creation nor validation.

If we are handed the credentials at subscription-time, then that credentials is "PLAIN" I think.

Reply:
@AxelNennker Agree that CloudEvents does not specify the behaviour of sinkCredential

Think, I get your point about the credentials at subscription-time.

About your suggestion of using "PLAIN" for now. What is your understanding of this model?
My understanding is API Consumer indicates identifier and secret at subscription-time and Telco Operator generates  HTTP Basic Auth to secure the notification sent to the API Consumer


- About ACCESSTOKEN

Being compliant with current proposal, it only makes sense this Token is generated by API Consumer on their own at subscription-time, without any interaction with Telco Operator. API consumer establishes _accessTokenExpiresUtc_ so as it controls its lifetime and validity.

Another point of discussion is what happens when the Token expires. The subscription is enabled but when Telco Operator indicates this token towards API Consumer it will reject the notification. 


The model about Telco Operator requesting a Token to an API Consumer (case sinkCredential not provided at subscription-time but is required to be secured) does not make sense to me (think your view is the same) because that implictly means we are requesting API Consumers to implement an AZ Server and also Telco Operators to interact with it. Too complex.